### PR TITLE
bpo-44562: Use PyType_GenericAlloc in Py_GenericAlias

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue44562

### CODE REVIEW

File: Objects/genericaliasobject.c [hunk 1]
```diff
@@ -646,11 +646,11 @@ PyTypeObject Py_GenericAliasType = {
 PyObject *
 Py_GenericAlias(PyObject *origin, PyObject *args)
 {
-    gaobject *alias = PyObject_GC_New(gaobject, &Py_GenericAliasType);
+    gaobject *alias = (gaobject*) PyType_GenericAlloc(
```
Comment (serhiy-storchaka) [inline Objects/genericaliasobject.c:649]: `PyType_GenericAlloc()` calls `_PyObject_GC_Malloc()` instead of `PyObject_GC_New()`.

Comment (pablogsal) [reply Objects/genericaliasobject.c:649]: `_PyObject_GC_New` calls `_PyObject_GC_Malloc` + `_PyObject_Init`, which `PyType_GenericAlloc` also calls. What I am missing?

Comment (serhiy-storchaka) [reply Objects/genericaliasobject.c:649]: All right then.

Please fix also `ga_new()`.

File: Objects/genericaliasobject.c [hunk 2]
```diff
@@ -646,11 +646,11 @@ PyTypeObject Py_GenericAliasType = {
...
+            (PyTypeObject *)(&Py_GenericAliasType), 0);
```
Comment (serhiy-storchaka) [inline Objects/genericaliasobject.c]:
```suggestion
            (PyTypeObject *)&Py_GenericAliasType, 0);
```

File: Objects/genericaliasobject.c (hunk 1 above)
Comment (Fidget-Spinner) [reply Objects/genericaliasobject.c:649]: ``ga_new`` doesn't seem to need the fix. It uses ``tp_alloc`` which is set to ``PyType_GenericAlloc ``.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**